walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,56 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> availableMeetingTimes = new ArrayList<TimeRange>();
+
+    if (request.getAttendees().size() == 0) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    if (events.size() == 0) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    int prevTime = TimeRange.START_OF_DAY;
+    for (Event event : events) {
+      final String attendeeName = event.getAttendees().iterator().next();
+      if (!request.getAttendees().contains(attendeeName)) {
+        continue;
+      }
```

Comment:
It looks like you're skipping when an attendeeName isn't in the required attendee's list. But you're only checking the first attendee.

This will fail if the first attendee of the event isn't required, but a later attendee is required.
```
Event("Event", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
    Arrays.asList(PERSON_A, PERSON_B)));
new MeetingRequest(Arrays.asList(PERSON_B), DURATION_60_MINUTES);
```

Reply:
Created a function that checks if event attendees contains any of the names in request attendees